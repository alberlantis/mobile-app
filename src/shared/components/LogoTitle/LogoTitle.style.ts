import { StyleSheet } from "react-native";

import colors from "src/theme/colors";

export default StyleSheet.create({
  container: {
    justifyContent: "center",
    alignItems: "center",
  },
  image: { width: 80, height: 80 },
  title: {
    color: colors.WHITE,
    fontWeight: "bold",
    fontSize: 28,
    marginTop: 10,
  },
});
