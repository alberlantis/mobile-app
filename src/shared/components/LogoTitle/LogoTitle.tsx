import React from "react";
import { Image, Text, View, Platform, StyleSheet } from "react-native";

import { useImageAssets } from "src/shared/hooks";
import s from "./LogoTitle.style";

interface ILogoTitleProps {
  title: string;
}

const LogoTitle: React.FC<ILogoTitleProps> = ({ title }) => {
  const { images } = useImageAssets();

  return (
    <View
      testID="logo-title-container-id"
      style={StyleSheet.compose(s.container, {
        marginTop: Platform.OS === "android" ? 56 : 0,
      })}
    >
      <Image
        source={images.logo}
        style={s.image}
        testID="logo-title-image-id"
      />
      <Text style={s.title}>{title}</Text>
    </View>
  );
};

export default LogoTitle;
