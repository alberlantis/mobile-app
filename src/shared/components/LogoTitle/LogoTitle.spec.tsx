import { render, screen } from "@testing-library/react-native";
import { Platform } from "react-native";

import LogoTitle from "./LogoTitle";
import s from "./LogoTitle.style";

jest.mock("src/shared/hooks", () => ({
  useImageAssets: () => ({
    images: {
      logo: "logo_image",
    },
  }),
}));

describe("LogoTitle", () => {
  describe("general functionality", () => {
    beforeEach(() => {
      Platform.OS = "ios";
      render(<LogoTitle title="test-title" />);
    });

    it("should match snapshot", () => {
      expect(screen.toJSON()).toMatchSnapshot();
    });

    it("should render logo image", () => {
      const image = screen.getByTestId("logo-title-image-id");
      expect(image).toBeOnTheScreen();
      expect(image).toHaveProp("source", "logo_image");
      expect(image).toHaveStyle(s.image);
    });

    it("should render logo without top margin when iOS", () => {
      const container = screen.getByTestId("logo-title-container-id");
      expect(container).toBeOnTheScreen();
      expect(container).toHaveStyle({
        ...s.container,
        marginTop: 0,
      });
    });
  });

  describe("when is android", () => {
    beforeEach(() => {
      Platform.OS = "android";
      render(<LogoTitle title="test-title" />);
    });

    it("should match snapshot", () => {
      expect(screen.toJSON()).toMatchSnapshot();
    });

    it("should render logo with top margin", () => {
      const container = screen.getByTestId("logo-title-container-id");
      expect(container).toBeOnTheScreen();
      expect(container).toHaveStyle({
        ...s.container,
        marginTop: 56,
      });
    });
  });
});
