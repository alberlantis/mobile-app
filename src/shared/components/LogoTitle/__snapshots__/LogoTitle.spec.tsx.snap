// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogoTitle general functionality should match snapshot 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "justifyContent": "center",
      },
      {
        "marginTop": 0,
      },
    ]
  }
  testID="logo-title-container-id"
>
  <Image
    source="logo_image"
    style={
      {
        "height": 80,
        "width": 80,
      }
    }
    testID="logo-title-image-id"
  />
  <Text
    style={
      {
        "color": "#FFF",
        "fontSize": 28,
        "fontWeight": "bold",
        "marginTop": 10,
      }
    }
  >
    test-title
  </Text>
</View>
`;

exports[`LogoTitle when is android should match snapshot 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "justifyContent": "center",
      },
      {
        "marginTop": 56,
      },
    ]
  }
  testID="logo-title-container-id"
>
  <Image
    source="logo_image"
    style={
      {
        "height": 80,
        "width": 80,
      }
    }
    testID="logo-title-image-id"
  />
  <Text
    style={
      {
        "color": "#FFF",
        "fontSize": 28,
        "fontWeight": "bold",
        "marginTop": 10,
      }
    }
  >
    test-title
  </Text>
</View>
`;
